### ISSUE TITLE
Controller latency over 10ms (bluetooth)

### ISSUE THREAD

**Issue by YumanC** [Ryochan7/DS4Windows#522]
This issue has given me a headache. DS4Windows displays my controller latency is over 10ms and the lightbar starts flashing red, causing input lag in games. I previously had a Gigabyte GC-WB867D-I REV Bluetooth 4.2 adapter, then bought a Plugable USB Bluetooth 4.0 adapter thinking the previous one was faulty. The latency became worse! (waste of money). I play 1.5-2 meters away from my PC, but my previous adapter had an antenna. I already turned off the option to allow the pc to turn off the adapter and save energy. What on earth could be causing this latency issue and how can I fix it?

I am using DS4windows version 1.6.0

**Comment by tremendoza** [comment]
Had the same issue and experimented with different scenarios to track it down. For me several things affected the connection:

USB Bandwidth  - Having multiple devices on the same root hub as the Bt Device. Case Front USB ports with poor power delivery (can't power a 2.5" External Drive) usually had issues as well.
 
Blockage - Objects between the Controller an BT Device, sometimes as little as my hand or folding my legs. Rear USB ports were almost unusable.

BT Device - Some devices just simply can't sustain the required bandwidth.

I solved these issues by modifying the antenna on my BT device once but still found that it wasn't consistent enough for me so I bought the Official Sony Adapter.

**Comment by Ryochan7** [comment]
There is not much that I can do to help with BT issues. One thing to mention is that you want to make sure that no other device is connected to your BT adapter while testing the connection. Make sure that the BT adapter is not in discovery mode. Also, you might want to play with the BT Poll Rate setting in a DS4Windows profile and see if using a different setting helps; the default is 4 ms which should be a fairly conservative value.

The version of Windows that you are using might cause some additional issues. Many people that have reported problems in the past have been running Windows 7. Some specific Windows 7 issues have been addressed such as device hotplugging issues. The MS Bluetooth Stack offered with Windows 7 is not as good as the version available for Windows 10 although the Windows 10 version isn't as good as the old Toshiba BT Stack. I have done some minor gameplay testing using Windows 7 and BT and I did not encounter any connection problems with either of my BT adapters.
